src/trackers/PillarboxMonitoring.js

```diff
@@ -0,0 +1,898 @@
+    this.player.off('playing', this.playbackStart);
+    this.player.off('pause', this.playbackStop);
+    this.player.off('error', this.error);
+    this.player.off(['playerreset', 'dispose', 'ended'], this.sessionStop);
+    this.player.off(['waiting', 'stalled'], this.stalled);
+
+    window.removeEventListener('beforeunload', this.sessionStop);
+  }
+
+  /**
+   * Remove the token from the asset URL.
+   *
+   * @param {string} assetUrl The URL of the asset
+   *
+   * @returns {string|undefined} The URL without the token, or undefined if the
+   *                             input URL is invalid
+   */
+  removeTokenFromAssetUrl(assetUrl) {
+    if (!assetUrl) return;
+
+    try {
+      const url = new URL(assetUrl);
+
+      url.searchParams.delete('hdnts');
+
+      return url.href;
+    } catch (e) {
+      return;
+    }
+  }
+
+  /**
+   * Resets the playback session and clears relevant properties.
+   *
+   * @param {Event} event The event that triggered the reset. If the event type
+   *                 is not 'ended' or 'playerreset', listeners will be removed.
+   */
+  reset(event) {
+    this.currentSessionId = undefined;
+    this.lastPlaybackDuration = 0;
+    this.lastPlaybackStartTimestamp = 0;
+    this.lastStallCount = 0;
+    this.lastStallDuration = 0;
+    this.loadStartTimestamp = 0;
+    this.metadataRequestTime = 0;
+    this.mediaAssetUrl = undefined;
+    this.mediaId = undefined;
+    this.mediaMetadataUrl = undefined;
+    this.mediaOrigin = undefined;
+    this.sessionStartTimestamp = undefined;
+    this.tokenRequestTime = 0;
+
+    clearInterval(this.heartbeatIntervalId);
+
+    if (event && !['ended', 'playerreset'].includes(event.type)) {
+      this.removeListeners();
+    }
+  }
+
+  /**
+   * Sends an event to the server using the Beacon API.
+   *
+   * @param {string} eventName Either START, STOP, ERROR, HEARTBEAT
+   * @param {Object} [data={}] The payload object to be sent. Defaults to an
+   *                           empty object if not provided
+   */
+  sendEvent(eventName, data = {}) {
+    // If the tracker is disabled for the current session, and there has been no
+    // previous session, no event is sent. However, if a session was already
+    // active, we still want to send the STOP event so that it is properly
+    // stopped.
+    if (
+      (this.isTrackerDisabled() && !this.currentSessionId) ||
+      !this.currentSessionId
+    ) return;
+
+    const payload = JSON.stringify({
+      event_name: eventName,
+      session_id: this.currentSessionId,
+      timestamp: PillarboxMonitoring.timestamp(),
+      version: this.schemaVersion,
+      data
+    });
+
+    navigator.sendBeacon(
+      this.beaconUrl,
+      payload
+    );
+  }
+
+  /**
+   * Starts a new session by first stopping the previous session, then resetting
+   * the session start timestamp and media ID to their new values.
+   */
+  sessionStart() {
```

Comment:
I would suggest a name here too, I find it inconsistent that:

- `sessionStop`: Sends the stop event and resets the state of the session.
- `sessionStart`: Starts tracking the session but doesn't send the start event.

I would suggests something like `initSessionTracking` or `initMetricsTracking`.